tools/jenerator/src/go.ml

```diff
@@ -0,0 +1,240 @@
+;;
+
+let gen_def service func args return_type =
+  let arg_list = List.map (fun a -> a.field_name ^ " " ^ gen_type a.field_type) args in
+  "func (c *" ^ service ^ "Client) " ^
+    snake_to_go_ident func ^ "(" ^
+      String.concat ", "
+                    arg_list ^
+      ") " ^ return_type ^ " {"
+;;
+
+let gen_constructor service =
+  let camel_name = snake_to_go_ident service in
+  [ (0, "func New" ^ camel_name ^ "Client(host string, name string) (*" ^
+          camel_name ^ "Client, error) {");
+	  (2,   "conn, err := net.Dial(\"tcp\", host)");
+	  (2,   "if err != nil {");
+		(4,   "return nil, err");
+	  (2,   "}");
+	  (2,   "mh := new(codec.MsgpackHandle)");
+	  (2,   "mh.StructToArray = true");
+	  (2,   "rpcCodec := codec.MsgpackSpecRpc.ClientCodec(conn, mh)");
+	  (2,   "client := rpc.NewClientWithCodec(rpcCodec)");
+	  (2,   "return &" ^ camel_name ^ "Client{*client, name}, nil");
+    (0, "}");
+  ]
+;;
+
+let gen_return_type m =
+  Option.default "" (Option.map gen_type m.method_return_type)
+;;
+
+let gen_client_method service_name m =
+  let name = m.method_name in
+  [ (0, gen_def (snake_to_go_ident service_name) name m.method_arguments (gen_return_type m));
+    (2,   gen_result m);
+    (2,   gen_client_call m);
+    (2,   "return result");
+    (0, "}");
+  ]
+;;
+
+let gen_client s =
+  let gen_client_with_service = gen_client_method s.service_name in
+  let methods = List.map gen_client_with_service s.service_methods in
+  let content = methods in
+  concat_blocks content;
+;;
+
+
+let gen_self_with_equal field_names =
+  List.map (fun s -> (0, "@" ^ s ^ " = " ^ s)) field_names
+;;
+
+let gen_str name field_names =
+  List.concat [
+    [
+      (0, "def to_s");
+      (1,   "gen = Jubatus::Common::MessageStringGenerator.new");
+      (1,   gen_call "gen.open" [gen_string_literal name])
+    ];
+    List.map (fun f ->
+      (1,   gen_call "gen.add" [gen_string_literal f; "@" ^ f])
+    ) field_names;
+    [
+      (1,   "gen.close()");
+      (1,   "return gen.to_s");
+      (0, "end")
+    ]
+  ]
+;;
+
+
+let gen_message m =
+  let fields = List.map (fun f -> (f.field_name, f.field_type) ) m.message_fields in
+  List.concat [
+      [(0, "type " ^ snake_to_go_ident m.message_name ^ " struct {")];
+      List.map (fun (name, type_name) ->
+                (2, snake_to_go_ident name ^ " " ^ gen_type type_name) ) fields;
+      [(0, "}")]
+  ]
+;;
+
+let gen_typedef = function
+  | Message m ->
+     gen_message m
+  | _ ->
+     [(0, "")]
+;;
+
+let gen_client_file conf source services =
+  let base = File_util.take_base source in
+  let filename = Filename.concat base "client.go" in
+  let clients = List.map (fun s ->
+    concat_blocks [
+      gen_client_struct s.service_name;
+      gen_constructor s.service_name;
+      gen_client s;
+    ]
+  ) services in
+
+  let content = concat_blocks [
+    [ (0, "package jubatus_client")];
+    [
+      (0, "import (");
+      (2, "common \"../common\"");
```

Comment:
I think it should be `github.com/...` path because user cannot load relative path without cloning and copying the common dir or put source to jubatus-go-client directory.